File: samplerxjavaapp/src/main/java/com/yoox/samplerxjavaapp/detail/ColorAdapter.java
package com.yoox.samplerxjavaapp.detail;

import android.content.Context;
import android.graphics.drawable.ColorDrawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.yoox.net.models.outbound.Color;
import com.yoox.samplerxjavaapp.R;

import java.util.List;

public class ColorAdapter extends RecyclerView.Adapter<ColorAdapter.ViewHolder> {

    private final List<Color> data;
    private final OnColorSelectedListener onItemSelectedListener;

    ColorAdapter(List<Color> data, OnColorSelectedListener onItemSelectedListener) {
        this.data = data;
        this.onItemSelectedListener = onItemSelectedListener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);

        View contactView = inflater.inflate(R.layout.item_color, parent, false);

        return new ViewHolder(contactView);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(data.get(position), onItemSelectedListener);
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    @Override
    public long getItemId(int position) {
        return data.get(position).hashCode();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        private final ImageView image;
        private final TextView name;

        ViewHolder(View itemView) {
            super(itemView);

            image = itemView.findViewById(R.id.image);
            name = itemView.findViewById(R.id.name);
        }

        void bind(Color item, OnColorSelectedListener onItemSelectedListener) {
            int color = android.graphics.Color.parseColor("#" + item.getRgb());
            image.setImageDrawable(new ColorDrawable(color));
            name.setText(item.getName());
            itemView.setOnClickListener(v -> onItemSelectedListener.onColorSelected(item));
        }
    }

    public interface OnColorSelectedListener {
        void onColorSelected(Color item);
    }

}

File: samplekotlinapp/src/main/java/com/yoox/samplekotlinapp/detail/ColorAdapter.kt
package com.yoox.samplekotlinapp.detail

import android.graphics.drawable.ColorDrawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.yoox.net.models.outbound.Color
import com.yoox.samplekotlinapp.R

class ColorAdapter internal constructor(
    private val data: List<Color>,
    private val onItemSelectedListener: (Color) -> Unit
) : RecyclerView.Adapter<ColorAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val context = parent.context
        val inflater = LayoutInflater.from(context)
        val contactView = inflater.inflate(R.layout.item_color, parent, false)

        return ViewHolder(contactView)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) =
        holder.bind(data[position], onItemSelectedListener)

    override fun getItemCount(): Int = data.size

    override fun getItemId(position: Int): Long = data[position].hashCode().toLong()

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val image: ImageView = itemView.findViewById(R.id.image)
        private val name: TextView = itemView.findViewById(R.id.name)

        fun bind(item: Color, onItemSelectedListener: (Color) -> Unit) {
            val color = android.graphics.Color.parseColor("#" + item.rgb)
            image.setImageDrawable(ColorDrawable(color))
            name.text = item.name
            itemView.setOnClickListener { onItemSelectedListener(item) }
        }
    }
}
